recognize jshell files

Konfekt opened the pull request:
Maybe @zzzyxwvut could have a look as well

Konfekt committed:
recognize jshell files

@zzzyxwvut

Konfekt committed:
highlight shebang in Jshell as comment

zzzyxwvut commented on runtime/ftplugin/java.vim:
```diff
@@ -78,6 +78,10 @@ setlocal comments& comments^=:///,sO:*\ -,mO:*\ \ ,exO:*/
 setlocal commentstring=//\ %s
 
+if expand('%:e') ==# 'jsh'
+  match Comment /^#!.*/
```
Please can you limit the match to some arbitrary top lines,  
e.g. `\%<32l^#!.*`, and make the indentation conform to the  
modeline?

Konfekt committed:
hash shebangs unsupported

To make Jshell scripts executable, use triple slash instead of a
shebang, see [these comments][0]
and  [this answer][1] for
executable jshell scripts. So no shebang detection needed as

```vim
if expand('%:e') ==# 'jsh'
    " See https://stackoverflow.com/a/44925456/3528522
    match Comment /\%<2l^\/\/\/.*/
endif
```

is redundant, since `//` is already a comment marker

See [2]

[0]: https://unix.stackexchange.com/a/162535
[1]: https://stackoverflow.com/a/44925456/3528522
[2]: https://github.com/vim/vim/pull/16260
